src/components/generic/Loading.tsx

```diff
@@ -56,25 +56,82 @@ const DotGroup = styled.div`
   display: flex;
 `;
 
+const PlayButton = styled.div`
+  width: 50px;
+  height: 50px;
+  background-color: #e0e0e0;
+  border: 2px solid #ccc;
+  border-radius: 50%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  font-size: 0.5rem;
+  font-weight: bold;
+  color: #333;
+  cursor: pointer;
+  transition: background-color 0.3s ease, transform 0.3s ease;
+
+  &:hover {
+    background-color: #d0d0d0;
+    transform: scale(1.05);
+  } 
+  &:active {
+    background-color: #c0c0c0;
+  }
+`;
+
+const DisplayWindow = styled.div`
+  width: 160px;
+  height: 80px;
+  gap: 100px;
+  border-radius: 20px;
+  background-color: #e0e0e0;
+  border: 2px solid #ccc;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  font-size: 1.5rem;
+  font-weight: bold;
+  color: #333;
+  margin-bottom: 20px;
+`;
+
+const ActivityButtonContainer = styled.div`
+  display: flex;
+  gap: 20px;
+  margin-bottom: 20px;
+  flex-direction: row;
+`;
+
+const ScreenContainer = styled.div`
+  display: flex;
+  margin-bottom: 40px;
+  flex-direction: column;
+`;
+
 const Loading = ({
-  size = "medium",
-  color = primaryColor,
+    size = 'medium',
+    color = primaryColor,
 }: {
-  size: Size;
-  color: string;
+    size: Size;
+    color: string;
 }) => {
-  return (
-    <Container>
-      <DotGroup>
-        <Dot size={sizeMapping[size]} color={color} />
-        <Dot size={sizeMapping[size]} color={color} />
-      </DotGroup>
-      <DotGroup>
-        <Dot size={sizeMapping[size]} color={color} />
-        <Dot size={sizeMapping[size]} color={color} />
-      </DotGroup>
-    </Container>
-  );
+    return (
+        <Container>
+            <DotGroup>
+                <Dot size={sizeMapping[size]} color={color} />
+                <Dot size={sizeMapping[size]} color={color} />
+            </DotGroup>
+            <DotGroup>
+                <Dot size={sizeMapping[size]} color={color} />
+                <Dot size={sizeMapping[size]} color={color} />
+            </DotGroup>
+        </Container>
+    );
 };
 
+Loading.ActivityButtonContainer = ActivityButtonContainer;
```

Comment:
I'm a bit confused about the structure here. The components defined in this Loading.tsx file are not necessarily related to Loading but are rather standalone components.